atom/epistula/epistula.py

```diff
@@ -86,23 +98,27 @@ def verify_signature(
             timestamp: Message timestamp
             uuid: Message UUID
             signed_by: Sender's address
-            signed_for: Optional receiver's address
-            now: Current timestamp (defaults to current time if not provided)
+            signed_for: Receiver's address (optional)
+            now: Current timestamp (defaults to current time if not provided) in seconds
 
         Returns:
             None if verification succeeds, error message string if it fails
         """
-        # Input validation
-        if not isinstance(signature, str):
-            return "Invalid signature type"
-        if not isinstance(signed_by, str):
-            return "Invalid sender key type"
-        if not isinstance(uuid, str):
-            return "Invalid UUID type"
-        if not isinstance(body, bytes):
-            return "Body is not of type bytes"
-        if signed_for is not None and not isinstance(signed_for, str):
-            return "Invalid receiver key type"
+
+        #Pydantic will enforce the validation typing rules
+        try:
+            VerifySignatureRequest(
+                body=body,
+                timestamp=timestamp,
+                signature=signature,
+                uuid=uuid,
+                signed_by=signed_by,
+                signed_for=signed_for,
+                now=now,
+            )
+
+        except ValidationError as e:
+            return f"Validation Error: {str(e)}"
 
         try:
             timestamp = int(timestamp)
```

Comment:
I also don't know why you need this.... Is it because sometimes it's a string? I think if we use pydantic we can remove this all together